Source.lua

```diff
@@ -181,7 +182,9 @@ local function luau_deserialize(bytecode, luau_settings)
 		luau_validatesettings(luau_settings)
 	end
 
-	local stream = buffer_fromstring(bytecode)
+	local isString = type(bytecode) == "string"
+
+	local stream = if isString then buffer_fromstring(bytecode) else bytecode
```

Comment:
If you make the change at line 513, you should inline the type check to get rid of the otherwise useless variable.